File: 01.CSharpPartOne/04_ConsoleInputAndOutput/ConsoleInputAndOutput/06QuadraticEquation/QuadraticEquation.cs
using System;

/*Problem 6. Quadratic Equation

Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).
 * */
class QuadraticEquation
{
    static void Main(string[] args)
    {
        Console.Write("Enter the a coefficient: ");
        decimal a = decimal.Parse(Console.ReadLine());

        Console.Write("Enter the b coefficient: ");
        decimal b = decimal.Parse(Console.ReadLine());

        Console.Write("Enter the c coefficient: ");
        decimal c = decimal.Parse(Console.ReadLine());

        decimal d = b * b - (4 * a * c);

        decimal x1 = 0m;
        decimal x2 = 0m;

        if (d < 0)
        {
            Console.WriteLine("no real roots");
        }
        else if (d == 0)
        {
            x1 = -(b / (2 * a));
            Console.WriteLine("x1 = x2 = {0}", x1);
        }
        else
        {
            x1 = -((b + (decimal)Math.Sqrt((double)d)) / (2 * a));
            x2 = -((b - (decimal)Math.Sqrt((double)d)) / (2 * a));

            Console.WriteLine("x1 = {0}; x2 = {1}", x1, x2);
        }
    }
}

File: 06_JavaScript/03_ContitionalStatements/ConditionalStatements/06.QuadraticEquation/QuadraticEquation.js
var a;
var b;
var c;

var aValues = [2, -1, -0.5, 5];
var bValues = [5, 3, 4, 2];
var cValues = [-3, 0, -8, 8];

for (var i = 0; i < aValues.length; i++) {
    a = aValues[i];
    b = bValues[i];
    c = cValues[i];

    var discriminant = 0;
    discriminant = b * b - (4 * a * c);

    if (discriminant < 0) {
        console.log('no real roots');
        document.write('<p>no real roots</p>');
        break;
    }

    var sqrtDiscriminant = 0;
    sqrtDiscriminant = Math.sqrt(discriminant);
    var x1 = (-b - sqrtDiscriminant) / (2 * a);
    var x2 = (-b + sqrtDiscriminant) / (2 * a);

    console.log('x1 = ' + x1 + '; x2 = ' + x2);
    document.write('<p>' + 'x1 = ' + x1 + '; x2 = ' + x2 + '</p>');
}